torch/distributed/_state_dict_utils.py

```diff
@@ -411,16 +411,18 @@ def tensor_func(
             if pin_memory:
 
                 def unpin_memory(t):
-                    succ = torch.cuda.cudart().cudaHostUnregister(t.data_ptr())
+                    succ = int(torch.cuda.cudart().cudaHostUnregister(t.data_ptr()))
```

Comment:
what is the type returned by cudaHostUnregister?

if it's not always an int, i worried about this raising an exception

Reply:
Generally uses this native enum: https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html#group__CUDART__TYPES_1g3f51e3575c2178246db0a94a430e0038

which maps to int values that are all < 10k